reactor-core/src/test/java/reactor/core/scheduler/SchedulersTest.java

```diff
@@ -724,6 +729,56 @@ public void dispose() {
 		}
 	}
 
+	@Test
+	public void testScheduleDirectPeriodicallyCancellsSchedulerTask() throws Exception {
```

Comment:
(nitpick) typo in the test naming: `Cancells` should be `Cancels`

Reply:
@simonbasle Thank you.

I'm on a planned vacation for a week and will submit the requested fix around October 9th.
Basically, I'm going to copy-paste WorkerTask into PeriodicWorkerTask and strip-off extra logic in the same way it was done for PeriodicTask.

There are alternative implementations possible. For example, task/worker state and state manipulation logic could be extracted into the corresponding classes managing classes:
```
class TaskState/PeriodicTaskState {
    void complete() {}
    void dispose() {}
    boolean isDisposed() {}
}
```
SchedulerTask and WorkerTask could just delegate to the corresponding TaskState/PeriodicTaskState.

What do you think?

Reply:
I'm ok with duplication of code here as it is on the critical path and it could be beneficial to avoid additional classes and intermediate methods (e.g. to limit the noise in already deep stacktraces when an error occur). enjoy your vacation 😉 🏖